drivers/sensors/cirque_pinnacle.c

```diff
@@ -192,6 +197,53 @@ void cirque_pinnacle_tune_edge_sensitivity(void) {
     ERA_ReadBytes(0x0168, &temp, 1);
 }
 
+// Perform calibration
+void cirque_pinnacle_calibrate(void) {
+    uint8_t  calconfig;
+    uint16_t timeout_timer;
+
+    // CalConfig1 (Compensation)
+    // Bit 0: Calibrate, 1=calibrate, 0=complete
+    // Bit 1: Background Comp Enable, 1=enable, 0=disable
+    // Bit 2: NERD Comp Enable (No, I don't know what NERD means), 1=enable, 0=disable
+    // Bit 3: Track Error Comp Enable, 1=enable, 0=disable
+    // Bit 4: Tap Comp Enable, 1=enable, 0=disable
+    // Bit 6: Calibration Matrix Disable, 1=disabled, 0=enabled
+    RAP_ReadBytes(CALIBRATION_CONFIG_1, &calconfig, 1);
+    calconfig |= 0x01;
+    RAP_Write(CALIBRATION_CONFIG_1, calconfig);
+
+    // Calibration takes ~100ms according to GT-AN-090624, doubling the timeout just to be safe
+    timeout_timer = timer_read();
+    do {
+        RAP_ReadBytes(CALIBRATION_CONFIG_1, &calconfig, 1);
+    } while ((calconfig & 0x01) && (timer_elapsed(timeout_timer) <= 200));
+
+    cirque_pinnacle_clear_flags();
+}
+
+// Enable/disable cursor smoothing, smoothing is enabled by default
+void cirque_pinnacle_cursor_smoothing(bool enable) {
+    uint8_t feedconfig3;
+
+    // FeedConfig3 (Advanced feature flags)
```

Comment:
Is this register available in all trackpad versions (TM040040/TM035035/TM023023) and Pinnacle firmwares?
GT-AN-090620 does not list it at all (presumably hidden under `RESERVED`.

Reply:
FeedConfig3 register is listed in GT-AN-090625, which is stated as a companion document to GT-AN-090620. Same thing is stated for the ERA document (GT-AN-090623) and compensation document (GT-AN-090624).
Additionally, it is documented in Cirque's circle trackpad example code [here](https://github.com/cirque-corp/Cirque_Pinnacle_1CA027/blob/master/Additional_Examples/AnyMeas_Example/Pinnacle.h). I don't see any indication it would not be present for a particular version of circle trackpad.
Empirically, I've tested this register on TM035035 and TM040040; the effect is subtle on 40mm but more prominent on 35mm.